#! /usr/bin/env python

PACKAGE = 'pointcloud_preprocessor'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator ()
# def add (self, name, paramtype, level, description, default = None, min = None, max = None, edit_method = ""):
gen.add ("base_frame", str_t, 0, "base frame", "base_link")
gen.add ("unit_axis", str_t, 0, "Unit vector used to compare the inner product with the normal vector of the plane", "z")
gen.add("outlier_threshold", double_t, 0, "threshold to rmeove outliers", 0.01, 0.0, 0.5)
gen.add("max_iterations", int_t, 0, "the maximum iteration", 1000, 0, 10000)
gen.add("min_inliers", int_t, 0, "the minimum number on a edge", 5000, 0, 100000)
gen.add("min_points", int_t, 0, "the minimum number on a edge", 1000, 0, 100000)
gen.add("plane_slope_threshold", double_t, 0, "Threshold for slope to recognize as ground", 10.0, 0.0, 90.0)
gen.add("height_threshold", double_t, 0, "threshold to rmeove point cloud within height threshold from plane", 0.01, 0.0, 0.5)
gen.add ("voxel_size_x", double_t, 0, "voxel size", 0.04, 0.01, 10.0)
gen.add ("voxel_size_y", double_t, 0, "voxel size", 0.04, 0.01, 10.0)
gen.add ("voxel_size_z", double_t, 0, "voxel size", 0.08, 0.01, 10.0)
gen.add ("debug", bool_t, 0, "flag to output debug message", False)

exit (gen.generate (PACKAGE, "pointcloud_preprocessor", "RANSACGroundFilter"))
